### PR TITLE
Update typing comments and add py.typed for PEP-561 compliance

### PR DESCRIPTION
**Pull request by bryanforbes**

### CODE REVIEW

**Comment by liliserf** [comment]
This looks great! Any estimate on when it might make it in?

**Comment by txomon** [comment]
Hello people, is there any chance we can get this forward?

**Comment by sdispater** [comment]
Thanks!

File: pendulum/parser.py [hunk 1]
```diff
@@ -13,7 +14,7 @@
 from .tz import UTC
 
 
-def parse(text, **options):
+def parse(text, **options):  # type: (str, **typing.Any) -> str
```
Comment (elemoine) [inline pendulum/parser.py]: Is the return type really `str`? I am not familiar with the pendulum code base, but this looks suspicious to me.

Comment (schot) [reply pendulum/parser.py]: I believe the correct return type is `typing.Union[pendulum.Date, pendulum.Time, pendulum.Duration]`.

**Pull request merged**